extensions/scarb-snforge-test-collector/src/compilation.rs

```diff
@@ -0,0 +1,68 @@
+use anyhow::Result;
+use cairo_lang_sierra::program::{ProgramArtifact, VersionedProgram};
+use camino::Utf8Path;
+use once_cell::sync::Lazy;
+use rayon::prelude::{IntoParallelRefIterator, ParallelIterator};
+use serde::Serialize;
+
+use crate::compilation::test_collector::{collect_tests, ForkConfig, RawForkConfig, TestCase};
+use crate::crate_collection::{CrateLocation, TestCompilationTarget};
+use crate::metadata::LinkedLibrary;
+
+pub mod test_collector;
+
+static BUILTINS: Lazy<Vec<&str>> = Lazy::new(|| {
+    vec![
+        "Pedersen",
+        "RangeCheck",
+        "Bitwise",
+        "EcOp",
+        "Poseidon",
+        "SegmentArena",
+        "GasBuiltin",
+        "System",
+    ]
+});
+
+#[derive(Debug, Clone, Serialize, PartialEq)]
+pub struct CompiledTestCrate<T: ForkConfig> {
+    pub sierra_program: VersionedProgram,
+    pub test_cases: Vec<TestCase<T>>,
+    pub tests_location: CrateLocation,
+}
+
+pub(crate) type CompiledTestCrateRaw = CompiledTestCrate<RawForkConfig>;
+
+pub fn compile_tests(
+    targets: &Vec<TestCompilationTarget>,
+    linked_libraries: &[LinkedLibrary],
+    corelib_path: &Utf8Path,
+) -> Result<Vec<CompiledTestCrateRaw>> {
+    targets
+        .par_iter()
```

Comment:
Just to make sure, we are okay with using Rayon in scarb?

Reply:
yes, that's ok